tcms/testruns/views.py

```diff
@@ -623,39 +623,41 @@ def get(self, request, run_id):
         return render(request, self.template_name, context_data)
 
 
-@require_POST
-@permission_required('testruns.change_testexecution')
-def update_case_run_text(request, run_id):
+@method_decorator(permission_required('testruns.change_testexecution'), name='dispatch')
+class UpdateCaseRunTextView(View):
     """Update the IDLE cases to newest text"""
 
-    test_run = get_object_or_404(TestRun, run_id=run_id)
+    http_method_names = ['post']
 
-    if request.POST.get('case_run'):
-        executions = test_run.case_run.filter(pk__in=request.POST.getlist('case_run'))
-    else:
-        executions = test_run.case_run.all()
-
-    executions = executions.filter(status__name='IDLE')
-
-    count = 0
-    updated_executions = ''
-    for execution in executions:
-        latest_version = execution.case.history.latest().history_id
-        if execution.case_text_version != latest_version:
-            count += 1
-            updated_executions += '<li>%s: %s -> %s</li>' % (
-                execution.case.summary, execution.case_text_version, latest_version
-            )
-            execution.case_text_version = latest_version
-            execution.save()
+    def post(self, request, run_id):
+        test_run = get_object_or_404(TestRun, run_id=run_id)
+
+        if request.POST.get('case_run'):
+            executions = test_run.case_run.filter(pk__in=request.POST.getlist('case_run'))
+        else:
+            executions = test_run.case_run.all()
+
+        executions = executions.filter(status__name='IDLE')
+
+        count = 0
+        updated_executions = ''
+        for execution in executions:
+            latest_version = execution.case.history.latest().history_id
+            if execution.case_text_version != latest_version:
+                count += 1
+                updated_executions += '<li>%s: %s -> %s</li>' % (
+                    execution.case.summary, execution.case_text_version, latest_version
+                )
```

Comment:
Instead of collecting the results in a string variable use `messages.add_message()` in each loop.